### DISCUSSION TITLE
Repository history predating 2010 (pre-Git?)

### DISCUSSION THREAD

**Discussion by phil-blain** [Unidata/netcdf-c#2817] [General]
Hi, I was looking for the reason why `ncdump` does not print newlines in its output when it encounters them (e.g. in attribute values) when the file type is netcdf-4. I'm talking about this code in `ncdump/ncdump.c::pr_att_string`:

https://github.com/Unidata/netcdf-c/blob/0c6fd782513251fb21281930baa885628c4976b7/ncdump/ncdump.c#L438-L446

So I did `git log -L:pr_att_string:ncdump/ncdump.c` to get the history of the function, but this code predates 18f4bca3 (moving to trunk subdir, 2010-06-03) where `ncdump/ncdump.c` was added to the Git repository.

Is the history pre-2010 available somewhere ?

**Comment by akrherz** [comment]
> Is the history pre-2010 available somewhere ?

This [Unidata blog post](https://www.unidata.ucar.edu/blogs/news/entry/netcdf_source_repositories_moved_to) denotes the subversion legacy was retained, but I don't think the CVS one ever was nor have seen a copy of that in the wild, maybe my memory is failing.  Googling found this [repo directory](https://www.gfd-dennou.org/arch/ucar/netcdf/old/) with ancient netcdf stuff included, maybe that would help your investigation?

**Comment by WardF** [comment]
@akrherz is correct, the migration from CVS to subversion was before I joined Unidata; one of my earlier tasks was to manage the migration from `svn` to `github`.  What information are you looking for? I presume an explanation beyond the one provided in the comments re: `\n`.  @edwardhartnett is responsible for the implementation of netCDF-4 (Thanks Ed!), he may have some insight if you have additional questions.

**Reply by phil-blain** [reply]
Thanks, yes, I'm curious as to what was the problem that led to this change. "would create an extra string in a list of strings" is not so clear to me. So I was hoping that the SVN commit log might have more information :) 

I'm not sure if it is related, but if I use `cdo showattribute test.nc` on a file with newlines in an text attribute, I get a `SIGABRT` in glibc with "corrupted size vs. prev_size" ...

**Reply by phil-blain** [reply]
... that is, if it's a netcdf-4 file.

**Reply by phil-blain** [reply]
OK, it seems the `cdo` bug is fixed in the latest (2.3.0) version .

**Comment by DennisHeimbigner** [comment]
I think the argument is more or less this.
NetCDF-4 added a "string" type. Before that (netcdf-3 and variants) only had "char" type.
In the netcdf-3 case attribute values can be strings of characters that are concatenated to produce
the final sequence of character values. In this case, printed newlines would not affect the concatenation result.
For netcdf-4, a string type attribute's value is also a sequence of strings, but they are not concatenated.
So adding lines apparently causes the output to look like there is an extra string hence the count of the attribute's
values is one too big.
